src/sql/workbench/services/notebook/common/sql/sqlSessionManager.ts

```diff
@@ -183,6 +186,29 @@ class SqlKernel extends Disposable implements nb.IKernel {
 		}
 	}
 
+	private setConnectionPath(): void {
+		if (this._path) {
+			let prefix = getUriPrefix(this._path);
+			if (!prefix || prefix === uriPrefixes.connection) {
+				this._connectionPath = uriPrefixes.notebook.concat(this._path);
```

Comment:
It's fine to have two prefixes here?

Reply:
If it doesn't have a prefix, getUriPrefix returns uriPrefixes.default:

`               let index = ownerUri.indexOf('://');

		if (index > 0) {

			prefix = ownerUri.substring(0, index + 3);

		} else {

			return uriPrefixes.default;
		}
`